packages/sdk/src/hooks/use-async-dispatch.ts

```diff
@@ -0,0 +1,12 @@
+import { Dispatch } from 'redux';
+import { useDispatch } from 'react-redux';
+import { TSdkAction } from '../types';
+
+// Wraps `dispatch` and cast the return type to a Promise, as the middleware
+// returns a Promise.
+function useAsyncDispatch<Action extends TSdkAction, Data>() {
+  const dispatch = useDispatch<Dispatch<Action>>();
+  return (action: Action) => (dispatch(action) as unknown) as Promise<Data>;
```

Comment:
`dispatch(action) as unknown` <= can you share why/what the `as unknown` means here? Thanks :)

Reply:
It's to force a type cast. You cast it to `unknown/any` (I personally always try to avoid `any`) and then cast it again to whatever you need to.